File: server/Inventory.Api/Module/Inventory/Domain/Port/Repository/IProductRepository.cs
using Inventory.Api.Module.Inventory.Domain.Model;
using Inventory.Api.Module.Shared.Domain.Port.Repository;

namespace Inventory.Api.Module.Inventory.Domain.Port.Repository
{
	public interface IProductRepository
	{
		Task<bool> AddAsync(Product entity, int userId);
        Task<IEnumerable<Product>> GetAllAsync();
        Task<bool> UpdateAsync(Product entity, int userId);
        Task<bool> RemoveAsync(int productId, int userId);
		Task<bool> ChangeStockAsync(int productId, int quantity, int userId);
	}
}

File: client/src/module/inventory/domain/repository/IProductRepository.ts
import type { Product } from "../entity/Product";
import type { ProductCreationModel } from "../model/ProductCreationModel";

export interface IProductRepository {
	GetProducts(): Promise<Product[]>;
	AddProduct(product: ProductCreationModel): Promise<void>;
	UpdateProduct(productId: number, product: ProductCreationModel): Promise<void>;
	UpdateStock(productId: number, quantity: number, userId: number): Promise<void>;
	RemoveProduct(productId: number, userId: number): Promise<void>;
}